src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1007,22 +1007,31 @@ internal NamedTypeSymbol GetSpecialType(SpecialType typeId, DiagnosticBag diagno
         /// </summary>
         internal Symbol GetSpecialTypeMember(SpecialMember member, DiagnosticBag diagnostics, CSharpSyntaxNode syntax)
         {
-            var memberSymbol = Compilation.GetSpecialTypeMember(member);
-            if ((object)memberSymbol != null)
+            Symbol memberSymbol;
+            return TryGetSpecialTypeMember(this.Compilation, member, syntax, diagnostics, out memberSymbol)
+                ? memberSymbol
+                : null;
+        }
+
+        internal static bool TryGetSpecialTypeMember<TSymbol>(CSharpCompilation compilation, SpecialMember specialMember, CSharpSyntaxNode syntax, DiagnosticBag diagnostics, out TSymbol symbol) where TSymbol : Symbol
+        {
```

Comment:
Consider using Symbol rather than TSymbol. That is, change to GetSpecialTypeMember that returns a possibly null Symbol and have the caller cast if needed.